### ISSUE TITLE
Prefer "open" to file constructor

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#8]
Originally reported by: **Julian O (BitBucket: [oddthinking](http://bitbucket.org/oddthinking), GitHub: @oddthinking?)**

---

Chapter 2 (Built-in Functions) of the Python Standard Library documentation (at least in Python 2.7.2) states:

> When opening a file, it’s preferable to use open() instead of invoking the file constructor directly.

e.g. the following code is preferred:

```
with open("filename", "rb") as source_file:
```

to this:

```
with file("filename", "rb") as source_file:
```

I have no idea of the rationale of this rule, but I do know I have been violating it consistently!

To help people like me, please consider adding a new warning code for when the file constructor is used directly.

(I guess it should not trigger if in the __init__ method of a subclass of file, but that might be rare enough to ignore.)

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/8
